android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -276,6 +276,29 @@ class GeckoEngineSession(
         )
     }
 
+    /**
+     * See [EngineSession.requestPrintContent]
+     */
+    override fun requestPrintContent() {
+        geckoSession.didPrintPageContent().then(
```

Comment:
Note aside: at some point, consider if it's valuable to move these actions to a `PrintController` so we have all the print actions on that if we are going to add more in-browser printing APIs.

```kotlin
geckoSession.printController.saveAsPdf()
geckoSession.printController.didPrintPageContent()
geckoSession.printController.printPageContent()
// etc..
```

Reply:
I think this is a good idea, I have [this](https://github.com/mozilla-mobile/firefox-android/pull/2317) open right now for more context too. Now that you mention it, both use-cases are converging into the same pattern.

Reply:
I like this plan, I think a good opportunity to refactor into a controller will be [bug 1838437](https://bugzilla.mozilla.org/show_bug.cgi?id=1838437) or when I implement the print UI.

Reply:
Filed [bug 1840715](https://bugzilla.mozilla.org/show_bug.cgi?id=1840715)! (Print UI patch was getting kinda big for this refactor to be included).